dask/dataframe/accessor.py

```diff
@@ -137,16 +137,23 @@ def split(self, pat=None, n=-1, expand=False):
     def cat(self, others=None, sep=None, na_rep=None):
         from .core import Series, Index
 
+        valid_types = (Series, Index, pd.Series, pd.Index)
+
         if others is None:
-            raise NotImplementedError("x.str.cat() with `others == None`")
+            if not isinstance(self, valid_types):
+                raise TypeError("self must be Series/Index")
+            return self._series.map_partitions(
+            str_cat, meta=self._series._meta
+        ).compute().str.cat(sep=sep, na_rep=na_rep)
```

Comment:
I don't think this is ideal. I don't think this should `compute` here. I'd find it suprising for `.str.cat()` to be immediate but `.str.cat(others=...)` to be delayed.

If we want to support this, I think that with `others=None` we'd return a Delayed string.